File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character,Integer> map= new HashMap<>();
        map.put('I',1);
        map.put('V',5);
        map.put('X',10);
        map.put('L',50);
        map.put('C',100);
        map.put('D',500);
        map.put('M',1000);

        char[] c=s.toCharArray();

        int num=0;

        for(int i=c.length-1;i>=0;i--){
            if(i==c.length-1){
                num+=map.get(c[i]);
            }
            else if(map.get(c[i])<map.get(c[i+1])){
                num-=map.get(c[i]);
            }
            else {
                num+=map.get(c[i]);
            }
        }
        return num;
    }
}

File: 13-roman-to-integer/roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        map ={
            "I":1,
            "V":5,
            "X":10,
            "L":50,
            "C":100,
            "D":500,
            "M":1000,
        }
        

        ans=0
        prev=0

        for c in reversed(s):
            value = map[c]
            if value < prev:
                ans -= value
            else:
                ans += value
                prev = value

        return ans
            